reserved_keyword_checker/reserved_keyword_checker.py

```diff
@@ -0,0 +1,292 @@
+app and inspecting the models of both the locally defined models and third party
+installed apps.
+
+EXAMPLE USAGE:
+
+python reserved_keyword_checker.py --reserved_keyword_file reserved_keywords.yml --override_file overrides.yml --report_path reports
+
+> reserved_keywords.yml: a yaml file containing a mapping of tool (i.e. Snowflake) to a
+  list of reserved keywords for that technology
+> overrides.yml: a yaml file containing a list of model fields to ignore when scanning
+  for reserved keyword violations.
+> reports: a path in which to write a csv report of the violations found with this script
+"""
+
+import inspect
+import io
+import re
+import os
+import sys
+
+import click
+import django
+import yaml
+from django.db import models
+
+
+class Violation(object):
+    """
+    A Django model field name that is in conclict with a defined list of reserved keywords
+    """
+
+    def __init__(self, model, field_name, system, override=False):
+        self.model = model
+        self.field = field_name
+        self.system = system
+        self.override = override
+
+    def __str__(self):
+        return "{} conflict in {}:{}:{}.{}".format(
+            self.system,
+            self.model._meta.app_label,
+            self.module_name,
+            self.model._meta.concrete_model.__name__,
+            self.field
+        )
+
+    def report_string(self):
+        """
+        Generate a comma-separated string used for creating a report of the reserved
+        keywords that were found. Output should be returned with the following fields:
+        - system: the system/tool that has the restriction that triggered the violation
+        - app source: was the violation detected in a locally defined or third party installed
+          application
+        - app_name: name of the Django app that contains the violation
+        - module_name: the file path of the Python module containing the violation
+        - model_name: name of the Django model that contains the violation
+        - field: the actual offending field name that triggered the violation
+        - source: is this module/model where the offending field was defined, or was it
+          inherited from another model.
+        """
+        model_name = self.model._meta.concrete_model.__name__
+        app_name = self.model._meta.app_label
+        if self.local_app:
+            app_source = "Local"
+        else:
+            app_source = "3rd party"
+        if not self.inherited:
+            source = "Defined here"
+        else:
+            source = "Inherited"
+        return "{},{},{},{},{},{},{}".format(
+            self.system, app_source, app_name, self.module_name, model_name, self.field, source
+        )
+
+    @property
+    def module_name(self):
+        """
+        The path to the module containing the reserved keyword violation
+        """
+        module_string = re.sub(r'\.', '/', self.model._meta.concrete_model.__module__)
```

Comment:
Nit: unnecessary use of `re` library.  
```suggestion
        module_string = self.model._meta.concrete_model.__module__.replace('.', '/')
```